src/runtime/cache.ts

```diff
@@ -78,11 +80,13 @@ export function defineCachedFunction<T = any>(
 
     const _resolve = async () => {
       if (!pending[key]) {
-        // Remove cached entry to prevent using expired cache on concurrent requests
-        entry.value = undefined;
-        entry.integrity = undefined;
-        entry.mtime = undefined;
-        entry.expires = undefined;
+        if (entry.value && !opts.allowStale) {
+          // Remove cached entry to prevent using expired cache on concurrent requests
+          entry.value = undefined;
+          entry.integrity = undefined;
+          entry.mtime = undefined;
+          entry.expires = undefined;
+        }
         pending[key] = Promise.resolve(resolver());
       }
       entry.value = await pending[key];
```

Comment:
Actually, this is not relevant for this PR but I just noticed that everything after the pending request is done should probably be done only once after the first time is it requested, no? Because right now it seems that if I request twice the same data, it is only resolving the promise once, but it is then setting `mtime`, `integrity`, validating and setting the result in the storage for each call; is it intended?

Reply:
Nice spot. However both value and time would be the same when promise resolves, we might move to to `Promise.resolve(resolver()).then(() => {...})` to do once

Reply:
Yes I was more thinking in terms of validating + setting the value in storage, which could be expensive especially if the storage driver is not local

Reply:
Indeed putting lock on remote storage to sync state would be expensive. We can however support later a flag that keeps initial time in promise and before writing to cache check the meta first and avoid write if there is a newer entity. considering cost at write.

Reply:
In the meantime couldn't we simply do something like:

```ts
const isPending = pending[key] // check whether there was a key (false the first time, true every other time)

if (!isPending) {
    ...
    pending[key] = Promise.resolve(resolver())
}

entity.value = await pending[key]

if (!isPending) {
    // do everything but only if there wasn't a key to begin with
}
```